src/rewrite.ml

```diff
@@ -40,27 +158,17 @@ let bind_match : pattern -> term -> (term, term) Bindlib.binder = fun p t ->
     | Patt(_,_,_) -> assert false
     | TEnv(_,_)   -> assert false
   in
-  Bindlib.unbox (Bindlib.bind_var x (lift_subst t))
+  Bindlib.unbox (Bindlib.bind_var x (lift_subst t2))
 
 (** [handle_rewrite t] rewrites according to the equality proved by [t] in the
     current goal. The term [t] should have a type corresponding to an equality
     (without any quantifier for now). All instances of the LHS are replaced by
     the RHS in the obtained goal. *)
 let handle_rewrite : term -> unit = fun t ->
-  (* Obtain the required symbols from the current signature. *)
```

Comment:
I like this version better, as this code will eventually be changed. In particular, `find_sym` is hackish, and it should not be used anywhere, so I'd rather it be local.